alica_engine/src/engine/RunningPlan.cpp

```diff
@@ -471,8 +472,27 @@ void RunningPlan::deactivate()
     } else {
         _ae->getTeamObserver().notifyRobotLeftPlan(_activeTriple.abstractPlan);
     }
+
     revokeAllConstraints();
     deactivateChildren();
+
+    /*
+     * Create instance of scheduler, later use shared instance for actual scheduling.
+     */
+    scheduler::Scheduler scheduler;
+
+    _terminateJob = make_shared<scheduler::Job>();
+    if(isBehaviour()) {
+        _terminateJob->cb = std::bind(&BasicBehaviour::onTermination, _basicBehaviour);
+    }
+    _terminateJob->isRepeated = false;
+    _terminateJob->repeatInterval = AlicaTime::microseconds(0);
+    std::weak_ptr<scheduler::Job> weakTerminateJob = _terminateJob;
+    scheduler.add(weakTerminateJob);
+
+    auto deactivatedSiblings = getDeactivatedSiblings();
+    std::cerr << "RUNNING PLAN DEACTIVATE: #siblings " << deactivatedSiblings.size() << std::endl;
```

Comment:
looks like you will remove it later and pass deactivatedSiblings termination jobs to _terminateJob->preRequisites